✨ Add 'append' property to write_files input

devholic opened the pull request:
**What this PR does / why we need it**:

Allow users to append contents to existing files.

**Which issue(s) this PR fixes**:

Fixes #6156

killianmuldoon commented on bootstrap/kubeadm/api/v1alpha3/kubeadmconfig_types.go:
```diff
@@ -197,6 +197,10 @@ type File struct {
 
+	// Append specifies whether to append Content to existing file if Path exists.
+	// +optional
+	Append bool `json:"append,omitempty"`
```
We should only change the latest version of the API - v1beta1.

killianmuldoon commented:
/ok-to-test

devholic commented:
/test pull-cluster-api-test-mink8s-main

sbueringer commented:
/lgtm

devholic commented:
Thanks for the review @killianmuldoon 🙂 

> Seems like a good change - have you got specific use cases you're looking to enable with this? It would be good to get an idea of what problems this is solving.

Similar to https://github.com/kubernetes-sigs/cluster-api/issues/6156, our team need to configure parameters / append to specific file at cloud-init because of the restriction of the environment (private cloud).

> I think you need to remove the changes in v1alpha3 and v1alpha4 and rerun the generations though. We should only add fields to the current versions of the APIs in general.

I removed changes from v1alpha3/v1alpha4 and added manual conversion functions 🙏

chrischdi commented:
/lgtm

killianmuldoon reviewed (approved):
/lgtm

sbueringer commented:
@devholic 
Thank you!

/lgtm
/approve
